lib/sparql_converter.js

```diff
@@ -0,0 +1,386 @@
+                order_results
+            );
+        } else {
+            let parameter = filters["name"];
+            let type = schema["_argmap"][parameter]["type"];
+            if (`${type}`.includes("Array")) {
+                let label_triplets = "";
+                let parameter_triplets = "";
+                let value = filters["value"]["value"];
+                let triplets = this.array_to_triplets(
+                    value,
+                    parameter,
+                    1,
+                    `${type}`
+                );
+
+                for (var i = 0; i < triplets[1].length; i++) {
+                    if (triplets[0][i] !== undefined)
+                        label_triplets += triplets[0][i];
+                    parameter_triplets += triplets[1][i];
+                }
+                return q.format(
+                    label_triplets,
+                    parameter_triplets,
+                    join_triplet,
+                    projection_triplet,
+                    sort_triplet,
+                    order_results
+                );
+            } else {
+                let triplets = this.filter_to_triplet(filters, schema, 1);
+                if (triplets[triplets.length - 1]) {
+                    return q.format(
+                        triplets[0],
+                        triplets[1],
+                        join_triplet,
+                        projection_triplet,
+                        sort_triplet,
+                        order_results
+                    );
+                } else {
+                    return q.format(
+                        "",
+                        triplets[0],
+                        join_triplet,
+                        projection_triplet,
+                        sort_triplet,
+                        order_results
+                    );
+                }
+            }
+        }
+    },
+
+    //converts filter to triplet
+    filter_to_triplet(filter_input, schema_input, index) {
+        const parameter = filter_input["name"];
+        const type = schema_input["_argmap"][parameter]["type"];
+
+        let value = filter_input["value"]["value"];
+        const isUnique = schema_input["_argmap"][parameter]["unique"];
+
+        if (isUnique) {
+            return [no_label_triplet.format(parameter, value), false];
+        } else if (`${type}` === "Date") {
+            return [
+                no_label_triplet.format(parameter, this.parse_date(value)),
+                false,
+            ];
+        } else if (`${type}` === "Number" || `${type}`.includes("Measure")) {
+            let formatted_value = `"%s"^^xsd:decimal`.format(value);
+            return [no_label_triplet.format(parameter, formatted_value), false];
+        } else if (`${type}`.includes("Entity")) {
+            return [no_label_triplet.format(parameter, "wd:" + value), false];
+        } else {
+            return [
+                get_label_triplet.format(index, value),
```

Comment:
How do you handle String-type parameters vs Entity parameters? Do you need escaping?